src/common.ts

```diff
@@ -0,0 +1,33 @@
+import { KeyStore } from 'keystore-idb/types'
+
+
+// CONSTANTS
+
+
+export const API_ENDPOINT = 'https://runfission.com'
+
+
+
+// BASE64
+
+
+export function base64UrlDecode(a: string): string {
+  return atob(a).replace(/\_/g, "/").replace(/\-/g, "+")
+}
+
+export function base64UrlEncode(b: string): string {
+  return makeBase64UrlSafe(btoa(b))
+}
+
+export function makeBase64UrlSafe(a: string): string {
+  return a.replace(/\//g, "_").replace(/\+/g, "-").replace(/=+$/, "")
+}
+
+
+
+// CRYPTO
+
+
+export function isRSAKeystore(ks: KeyStore): boolean {
```

Comment:
I'd maybe probably move this over `keystore/basic.ts`

I've been thinking of `common` as only having javascript helpers: encoding/decoding, typechecks, lodash-esque functions, etc

Also this should arguably be a function in the `keystore-idb` pkg

Reply:
Yeah, that makes sense 👍 

I'll add it there. The only reason I really need it is to differentiate between the types of key stores. Is there another way of doing this? I was thinking it'd be something along the lines of:
```
keystore.algorithm => RSA
```
and then we'd have a list somewhere with all the algorithms, or whatever you wanna call it.
Thoughts?

Reply:
Yeah this could be tightened up as 
```ts
return ks.cfg.type === 'rsa'
```

There is an enum in `keystore-idb` for this:
```ts
export enum CryptoSystem {
  ECC = 'ecc',
  RSA = 'rsa',
}
```

So a non-stringly typed version would be
```ts
import { CryptoSystem } from 'keystore-idb/types'

function isRSAKeystore(ks: KeyStore): boolean {
  return ks.cfg.type === CryptoSystem.RSA
}
```

We also might want to add in a type assertion for TS here. not really necessary but also doesn't hurt :
```ts
isRSAKeystore(ks: Keystore): ks is RSAKeyStore
```
(still returns a boolean, but also lets the TS Compiler know what type of KS it is)

Reply:
but yeah I'm actually leaning towards just putting this on the `KeystoreBase` class in `keystore-idb`:
```ts
  isRSA(): this is RSAKeyStore {
    return this.cfg.type === CryptoSystem.RSA
  }

  isECC(): this is ECCKeyStore {
    return this.cfg.type === CryptoSystem.ECC
  }
```